packages/codegen-ui/lib/__tests__/generate-form-definition/helpers/model-fields-configs.test.ts

```diff
@@ -313,6 +374,49 @@ describe('mapModelFieldsConfigs', () => {
     expect(formDefinition.elementMatrix).toStrictEqual([['ownerId']]);
   });
 
+  it('should not add nonModel field to matrix if nonModel not supported', () => {
+    const formDefinition: FormDefinition = getBasicFormDefinition();
+
+    const dataSchema: GenericDataSchema = {
+      dataSourceType: 'DataStore',
+      enums: {},
+      nonModels: { Interaction: { fields: {} } },
+      models: {
+        Dog: {
+          primaryKeys: ['id'],
+          fields: {
+            ownerId: {
+              dataType: { nonModel: 'Interaction' },
+              readOnly: false,
+              required: false,
+              isArray: false,
+            },
+          },
+        },
+      },
+    };
+
+    const modelFieldsConfigs = mapModelFieldsConfigs({ dataTypeName: 'Dog', formDefinition, dataSchema });
+
+    expect(modelFieldsConfigs).toStrictEqual({
+      ownerId: {
+        dataType: {
+          nonModel: 'Interaction',
+        },
+        inputType: {
+          isArray: false,
+          name: 'ownerId',
+          readOnly: false,
+          required: false,
+          type: 'TextAreaField',
+          value: 'ownerId',
+        },
+        label: 'Owner id',
+      },
+    });
+    expect(formDefinition.elementMatrix).toStrictEqual([]);
```

Comment:
curious about this assertion, is `mapModelFieldsConfigs()` expected to mutate the `formDefinition` that was passed in?

Reply:
Yes, `mapModelFieldsConfigs` mutates `formDefinition`